test/performance/range/view/view_take_benchmark.cpp

```diff
@@ -0,0 +1,174 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#include <deque>
+#include <list>
+#include <forward_list>
+#include <string>
+#include <vector>
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/range/view/take.hpp>
+#include <seqan3/range/view/take_exactly.hpp>
+#include <seqan3/range/view/single_pass_input.hpp>
+
+using namespace seqan3;
+
+// ============================================================================
+//  iterate_left_to_right
+// ============================================================================
+
+template <typename container_t, typename adaptor_t, bool single_pass = false>
+void iterate_left_to_right(benchmark::State & state)
+{
+    container_t c;
+    c.resize(1'000'000);
+    uint8_t i = 0;
+    for (auto & e : c)
+        e = ++i; // dummy values
+
+    unsigned dummy = 0;
+
+    // if single_pass, add view::single_pass_input, otherwise just &
+    using single_t = std::conditional_t<single_pass, decltype(c | view::single_pass_input), container_t &>;
+
+    if constexpr (std::Same<adaptor_t, void>)
+    {
+        for (auto _ : state)
+        {
+            single_t s{c};
```

Comment:
In the other benchmarks you do not benchmark the construction of the view, so you should probably not time this and exclude it via `state.PauseTiming()` and `state.ResumeTiming()` ?

Reply:
I need to recreate the view because of the single_pass case. Since the view creation is constant-time and 1million iterations after that I assume it won't influence it, but I can try with start/stopping, too.